scripts/test/jest.config.base.js

```
@@ -104,7 +104,9 @@ export const getBaseJestConfig = (isGlobal) => {
   const config = {
     ...baseConfig,
     setupFiles: [
-      '@finos/legend-dev-utils/jest/disallowConsoleError',
+      ...(options?.allowConsoleError
```

ag-grid enterprise outputs a bunch of messages to console.error when running without the license set, so I added this option so we can run tests and still permit console.error messages when we need it.

---

This is needed so `mobx` and `react` error can be caught since they don't throw, but just `console.error` out. I would say, we can put a `NODE_ENV === 'test'` to not render this grid or modify the condition of the `console.error` mutation to include `NODE_ENV === 'test'` as well, not just 'dev'.

https://github.com/finos/legend-studio/blob/4559d62a8eec7900490ad03df1fe5e399101c7ff/packages/legend-data-cube/src/components/view/grid/DataCubeGrid.tsx#L385

and 

https://github.com/finos/legend-studio/blob/4559d62a8eec7900490ad03df1fe5e399101c7ff/packages/legend-data-cube/src/components/view/grid/DataCubeGrid.tsx#L405

can be changed to `NODE_ENV !== 'production'` I think

---

Good call! I didn't realize we were already doing this to suppress the error logs in dev. This makes more sense so I implemented your suggestion.